lib/src/kbasesearchengine/events/handler/WorkspaceEventHandler.java

```diff
@@ -649,4 +678,143 @@ private static String toWSRefPath(final List<GUID> objectRefPath) {
         return String.join(";", refpath);
     }
 
+    @Override
+    public StatusEvent updateEvent(final StatusEvent ev)
+            throws IndexingException,
+                   RetriableIndexingException {
+
+        // only object level events are updated below
+        if (!ObjectEventQueue.OBJ_LVL_EVENTS.contains(ev.getEventType())) {
+            return ev;
+        }
+
+        // brute force get latest state and update event as event can be played
+        // out of order in the case of failed events being replayed.
+
+        // states to update
+        final StatusEventType latestEventType = ev.getEventType();
+        Boolean latestIsPublic = null;
+        String latestName = null;
+
+        if (ev.isPublic().isPresent()) {
+            latestIsPublic = ev.isPublic().get().booleanValue();
+        }
+        if (ev.getNewName().isPresent()) {
+            latestName = ev.getNewName().get().toString();
+        }
+
+
+        // wsid and objid of the object that the specified StatusEvent is an event for
+        final int wsid = ev.getAccessGroupId().get();
+        final long objid = Long.valueOf(ev.getAccessGroupObjectId().get().toString()).longValue();
+
+        Map<String, Object> command;
+
+        // check if workspace is permanently deleted or marked as deleted
+        try {
+            final Tuple9 tuple = getWorkspaceInfo(wsid);
+        } catch(IOException ex) {
+            throw new RetriableIndexingException(ErrorType.OTHER, ex.getMessage());
+        } catch(JsonClientException ex) {
+            // hacky: if exception message is changed, this logic will silently fail
+            if (ex.getMessage().contains("No workspace with id ")) {
+                return StatusEvent.
+                        getBuilder(ev.getStorageObjectType().get(), ev.getTimestamp(),
+                                StatusEventType.DELETE_ALL_VERSIONS).
+                        withNullableAccessGroupID(wsid).
+                        withNullableObjectID(Long.toString(objid)).build();
+            }
+        }
+
+        // check if object is permanently deleted or marked as deleted
+        command = new HashMap<>();
+        command.put("command", "listObjects");
+        command.put("params", new ListObjectsParams().
+                                         withIds(Arrays.asList((long)wsid)).
+                                         withMinObjectID(objid-1).
```

Comment:
spaces around operators

Reply:
Do you need the -1 and +1 at all? I'd assume that if the min and max were the same it'd return the exact object you want.

Reply:
```
In [7]: ws.list_objects({'ids': [26328], 'showHidden': 1, 'minObjectID': 2, 'maxObjectID': 2})
Out[7]: 
[[2,
  u'data_palette',
  u'DataPalette.DataPalette-1.0',
  u'2017-09-27T21:30:32+0000',
  4,
  u'gaprice',
  26328,
  u'gaprice:clonetestFrom24806',
  u'f07bfeb1a2cc031a04854959202c4e28',
  66,
  None]]
```

Reply:
Fixed.